metadata-io/src/main/java/com/linkedin/metadata/event/EventProducer.java

```diff
@@ -47,30 +48,35 @@ void produceMetadataAuditEvent(
    * @param urn the urn associated with the entity changed
    * @param aspectSpec aspect spec of the aspect being updated
    * @param metadataChangeLog metadata change log to push into MCL kafka topic
+   *
+   * @return A {@link Future} object that reports when the message has been produced.
    */
-  void produceMetadataChangeLog(
+  Future<?> produceMetadataChangeLog(
```

Comment:
I still need to read where these are getting used, but this is concerning to me. Currently we're operating under the assumption these are synchronous calls

Reply:
We have never used these Futures. This PR merely exposes that.